packages/gateway-client/src/worker/p2p-fetch/injectors.ts

```diff
@@ -56,23 +58,123 @@ const HEARTBEAT_SNIPPET = `<script>
     }
 </script>`;
 
-// this is the injector function that will be called for each response
-const injectSWHeartbeat: Injector = (headers, body) => {
-    // check if the response is html
-    if (headers.get('content-type')?.startsWith('text/html')) {
-        const [start, end] = body.toString().split(HEARTBEAT_SPLIT);
-        // check if the response contains the <head> tag
-        if (start && end) {
-            const parts = [start, HEARTBEAT_SPLIT, HEARTBEAT_SNIPPET, end];
-            const newBody = Buffer.from(parts.join(''));
-            // update the headers to have the correct content-length
-            headers.set('content-length', newBody.byteLength.toString());
-            return newBody;
+const makeInjector =
+    (content_type: string, split: string, snippet: string) =>
+    (headers: Headers, body: Buffer) => {
+        // check if the response is html
+        if (headers.get('content-type')?.startsWith(content_type)) {
+            const [start, end] = body.toString().split(split);
+            // check if the response contains the <head> tag
+            if (start && end) {
+                const parts = [start, split, snippet, end];
+                const newBody = Buffer.from(parts.join(''));
+                // update the headers to have the correct content-length
+                headers.set('content-length', newBody.byteLength.toString());
+                return newBody;
+            }
+        }
+
+        // if the response is not html or does not contain the <head> tag, return the original body
+        return body;
+    };
+
+const WEBSOCKET_CONTENT_TYPE = 'text/html';
```

Comment:
I think it is time to split this file up. We can follow the same pattern I used for `fetch-handlers`:
- New `p2p-fetch/injectors` directory
  - `Injectors` singleton goes into `injectors.ts` as default export
    - Define new exported `Injector` type (rename your current internal type to something else, i.e. `CompiledInjector`) with properties `contentType`, `split`, `snippet`.
    - `makeInjector()` becomes instance method on `Injectors` class, and can be renamed to `use(injector: Injector)`
  - Two new `webosocket-injector.ts` and `heartbeat-injector.ts` files that each export an `Injector` (via both named and default export)
- In `p2p-fetch/injdex.ts`:
  - Use your injectors:
 ```
    injectors
        .use(websocketsInjector)
        .use(heartbeatInjector)
  ```